packages/victory-core/src/victory-util/axis.js

```diff
@@ -209,13 +186,56 @@ function getTickArray(props) {
     ticks = stringMap ? tickValues.map((tick) => stringMap[tick]) : range(1, tickValues.length + 1);
   }
   const tickArray = ticks ? uniq(ticks) : getTicksFromFormat(props);
-  const filterArray = (arr) => {
+  const buildTickArray = (arr) => {
+    const newTickArray = [];
     const domain = (props.domain && props.domain[axis]) || props.domain;
-    return Array.isArray(domain)
-      ? arr.filter((t) => t >= Math.min(...domain) && t <= Math.max(...domain))
-      : arr;
+    if (arr) {
+      arr.forEach(function (t, index) {
+        if (Array.isArray(domain)) {
+          if (t >= Math.min(...domain) && t <= Math.max(...domain)) {
+            newTickArray.push({
+              value: t,
+              index
+            });
+          }
+        } else {
+          newTickArray.push({
+            value: t,
+            index
+          });
+        }
+      });
+      return newTickArray;
+    } else return undefined;
   };
-  return Array.isArray(tickArray) && tickArray.length ? filterArray(tickArray) : undefined;
+  return Array.isArray(tickArray) && tickArray.length ? buildTickArray(tickArray) : undefined;
+}
+
+function getTickFormat(props, scale) {
+  const { tickFormat } = props;
+  const axis = getAxis(props);
+  const stringMap = props.stringMap && props.stringMap[axis];
+  if (!tickFormat) {
+    const defaultTickFormat = getDefaultTickFormat(props);
+    const scaleTickFormat =
+      scale.tickFormat && isFunction(scale.tickFormat) ? scale.tickFormat() : (x) => x;
+    return defaultTickFormat || scaleTickFormat;
+  } else if (tickFormat && Array.isArray(tickFormat)) {
+    const tickArrayIndices = getTickArray(props)
```

Comment:
kind of wasteful to call `getTickArray` twice here. Prefer storing the value in a const, or checking with something less expensive like Array.isArray(props.tickValues)